homeassistant/components/alexa/handlers.py

```diff
@@ -1200,6 +1205,17 @@ async def async_api_set_mode(
             msg = f"Entity '{entity.entity_id}' does not support Mode '{mode}'"
             raise AlexaInvalidValueError(msg)
 
+    # Remote Activity
+    if instance == f"{remote.DOMAIN}.{remote.ATTR_ACTIVITY}":
+        activity = mode.split(".")[1]
+        activities: list[str] | None = entity.attributes.get(remote.ATTR_ACTIVITY_LIST)
+        if activity != PRESET_MODE_NA and activities and activity in activities:
+            service = remote.SERVICE_TURN_ON
+            data[remote.ATTR_ACTIVITY] = activity
```

Comment:
I think we should not use the mode controller to turn the remote on

Reply:
The turn on is also implemented for the power controller.
This code here sets the new activity. To use the remote's turn on service was the only way i found to set a new activity. Is there a better way?
Even though it looks a little bit strange i do think that it makes total sense to turn on the device of the remote when setting a new activity.

Reply:
I opened a discussion for this: https://github.com/home-assistant/architecture/discussions/1109.
Personally I do not think we should call service `TURN_ON` here.

https://github.com/home-assistant/architecture/discussions/1109